kernel/stack.c

```diff
@@ -97,14 +97,14 @@ int k_stack_cleanup(struct k_stack *stack)
 int z_impl_k_stack_push(struct k_stack *stack, stack_data_t data)
 {
 	struct k_thread *first_pending_thread;
-	k_spinlock_key_t key;
+	int32_t ret = 0;
```

Comment:
match type of function return value prototype and just use `int`

Reply:
Great point, thank you! I'll fix it shortly.